distributed-job-manager/src/main/kotlin/ru/fix/distributed/job/manager/Manager.kt

```diff
@@ -0,0 +1,87 @@
+package ru.fix.distributed.job.manager
+
+import org.apache.curator.framework.CuratorFramework
+import org.apache.curator.framework.recipes.cache.PathChildrenCache
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheEvent
+import org.apache.curator.framework.recipes.cache.PathChildrenCacheListener
+import org.apache.curator.framework.recipes.leader.LeaderLatch
+import org.apache.logging.log4j.kotlin.Logging
+import ru.fix.aggregating.profiler.Profiler
+import ru.fix.distributed.job.manager.model.DistributedJobManagerSettings
+
+/**
+ * Only single manager is active on the cluster.
+ * Manages job assignments on cluster by modifying assignment section of zookeeper tree.
+ *
+ * @author Kamil Asfandiyarov
+ * @see Worker
+ */
+class Manager(
+        curatorFramework: CuratorFramework,
+        profiler: Profiler,
+        settings: DistributedJobManagerSettings
+) : AutoCloseable {
+    private val paths = ZkPathsManager(settings.rootPath)
+    private val nodeId = settings.nodeId
+    private val workPoolCleanPeriod = settings.workPoolCleanPeriod
+
+    private val leaderLatchExecutor = LeaderLatchExecutor(
+            profiler, LeaderLatch(curatorFramework, paths.leaderLatch())
+    )
+    private val cleaner = Cleaner(
+            profiler, paths, curatorFramework, leaderLatchExecutor
+    )
+    private val rebalancer = Rebalancer(
+            profiler, paths, curatorFramework, leaderLatchExecutor, settings.assignmentStrategy, nodeId
+    )
+
+    private val workersAliveChildrenCache = PathChildrenCache(
```

Comment:
Let's replace with CuratorCache, since PathChildrenCache is depricated.
Also we should wait for cache to be initialized before calling
https://stackoverflow.com/questions/37825464/when-using-curator-treecache-how-can-i-ensure-cache-is-ready

Reply:
PathChildrenCache replaced with CuratorCache.
As far as I understand CuratorCache doesn't have analog of INITIALIZED event from PathChildrenCache.

Reply:
PathChildrenCache replaced by CuratorCache; 

Initialized event now handled before starting cleaner (which uses data from that cache) and rebalancer (to reduce excess rebalances)